src/lit-html.ts

```diff
@@ -330,6 +330,9 @@ export class NodePart extends Part {
         itemStart = itemEnd;
       }
       this._previousValue = itemParts;
+    } else if (this.startNode.nextSibling! === this.endNode.previousSibling! &&
+        this.startNode.nextSibling!.nodeType === Node.TEXT_NODE) {
+      this.startNode.nextSibling!.textContent = value;
```

Comment:
Can we make sure we have a test that updates an expression from a simple value to a complex one. I think it'll pass, but I don't see it covered yet. Something like:

```typescript
let value = 'foo';
const t = () => html`<div>${value}</div>`;
render(t(), container);
assert.equals(container.innerHTML, '<div>foo</div>');

value = html`<span>bar</span>`;
render(t(), container);
assert.equals(container.innerHTML, '<div><span>bar</span></div>');
```